runtime/near-vm-runner/src/cache.rs

```diff
@@ -3,16 +3,55 @@ use crate::prepare;
 use borsh::{BorshDeserialize, BorshSerialize};
 #[cfg(not(feature = "no_cache"))]
 use cached::{cached_key, SizedCache};
+use lazy_static::lazy_static;
 use near_primitives::hash::CryptoHash;
+use near_primitives::types::AccountId;
 use near_primitives::types::CompiledContractCache;
 use near_vm_errors::CacheError::{DeserializationError, ReadError, SerializationError, WriteError};
 use near_vm_errors::{CacheError, VMError};
 use near_vm_logic::{VMConfig, VMKind};
+use std::collections::HashMap;
 use std::convert::TryFrom;
+use std::sync::RwLock;
 use wasmer_runtime::{compiler_for_backend, Backend};
 use wasmer_runtime_core::cache::Artifact;
 use wasmer_runtime_core::load_cache_with;
 
+lazy_static! {
+    static ref ALWAYS_IN_MEM_CONTRACT: RwLock<HashMap<AccountId, Option<(CryptoHash, Result<wasmer_runtime::Module, VMError>)>>> = {
+        let mut m = HashMap::new();
+        m.insert("evm".to_string(), None);
+        // Add more always in memory contract account ids
+        RwLock::new(m)
+    };
+}
```

Comment:
Software-architecture wise, I am somewhat surprised that we use a global variable here, instead of storing this as a field of some kind of context struct. As we might want to have several runtime instances (for sharding), I fear that this'll bite us down the road. But this isn't specific to the PR, the `MODULES` below suffer from the same problem.

Reply:
Overall it doesn't look like proper long-term approach. 
First: the name is not proper contract identifier, content hash is.
Second: this kind of information belongs to configuration, not to the binary code.

Reply:
Yes it's problematic for multiple runner instance, existing code assumed there's only one runner

Reply:
As @matklad mentioned, it becomes a problem when we introduce parallel runtimes, which may be very soon

Reply:
Has been redo to keep in mem in NightshadeRuntime, and pass in as arguments